docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,448 @@
+---
+title: Transitioning Data Pipelines from Development to Production | Dagster
+description: This guide walks through how to transition your data pipelines from local development to production
+---
+
+# Transitioning Data Pipelines from Development to Production
+
+In this guide, we'll walk through how to transition your data pipelines from local development to staging and production deployments.
+
+Let's say we’ve been tasked with fetching the N most recent entries from Hacker News and splitting it into two datasets: one containing all of the data about stories and one containing all of the data about comments. We have two additional requirements:
+
+- We must be able to run our data pipeline in local, staging, and production environments.
+- We need to be confident that data won't be accidentally overwritten (for example because a user forgot to change a configuration value).
+
+Using a few Dagster concepts, we can easily tackle this task! If you’re new to Dagster, here’s an overview of the main concepts we’ll be using in this guide (if you’re already familiar with Dagster, feel free to skip to the next paragraph):
+
+- [Asset](/concepts/software-defined-assets) - an asset is a software object that models a data asset. The prototypical example is a table in a database or a file in cloud storage.
+- [Resource](/concepts/resource) - a resource is an object that models a connection to a (typically) external service. Resources can be shared between assets, and different implementations of resources can be used depending on environment. For example, a resource may provide methods to send messages in Slack.
+- [IO manager](/concepts/io-management/io-managers) - an IO manager is a special kind of resource that handles storing and loading assets. For example, if we wanted to store assets in S3, we would use Dagster’s built in S3 IO manager.
+- [Run config](/concepts/configuration/config-schema) - assets and resources sometimes require configuration to set certain values (like the password to a database). Run config allows you to set these values at run time. In this guide, we will also use an API to set some default run configuration.
+
+Using these Dagster concepts we will:
+
+- Write three assets: the full Hacker News dataset, data about comments, and data about stories.
+- Use Dagster's Snowflake IO manager to store the datasets in [Snowflake](https://www.snowflake.com/).
+- Set up our Dagster [repository](/concepts/repositories-workspaces/repositories) so that the configuration for the Snowflake IO manager is automatically supplied based on the environment where the code is running.
+
+<Note>
+  At the end of this guide, you'll find an optional section that demonstrates
+  writing a resource to fetch data from Hacker News and a pattern for mocking
+  resources that can be helpful when writing unit tests.
+</Note>
+
+### Setup
```

Comment:
```suggestion
---

## Setup
```
[1]
- Adding a horizontal rule to visually separate the section
- This should be an H2